lib/_http_client.js

```diff
@@ -22,6 +22,9 @@ function ClientRequest(options, cb) {
 
   if (typeof options === 'string') {
     options = url.parse(options);
+    if (!options.host && !options.hostname) {
```

Comment:
Why not just check `options.hostname`? Is there a reason for allowing empty hostnames at this point?
Example: `url.parse('http://:80/')`: `host: ':80', hostname: ''`, the only difference per doc between them is the presense of port in `host`.